### PR TITLE
add applyto tests for controller configs

### PR DESCRIPTION
**Pull request by aramase**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup

**What this PR does / why we need it**:
Adds a unit test to check the `ApplyTo` for controller configs. Having this test will ensure when we add new flags the config is correctly populated with the options. 

Following up on this PR fix comment - https://github.com/kubernetes/kubernetes/pull/85500#issuecomment-557189774

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
While writing this test I also found 2 missing flags in HPA for which the config is not set using the options.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/options_test.go [hunk 1]
```diff
@@ -390,6 +392,181 @@ func TestAddFlags(t *testing.T) {
 	}
 }
 
+func TestApplyTo(t *testing.T) {
+	fs := pflag.NewFlagSet("addflagstest", pflag.ContinueOnError)
+	s, _ := NewKubeControllerManagerOptions()
+	for _, f := range s.Flags([]string{""}, []string{""}).FlagSets {
```
Comment (liggitt) [inline cmd/kube-controller-manager/app/options/options_test.go]: comments on what these args are would be helpful

**Comment by liggitt** [comment]
one nit, lgtm otherwise. thanks

**Comment by liggitt** [comment]
needs `hack/update-bazel.sh` as well

File: cmd/kube-controller-manager/app/options/options_test.go (hunk 1 above)
Comment (aramase) [reply cmd/kube-controller-manager/app/options/options_test.go]: thanks for the review @liggitt! I've added a generic comment about these args being required for kube controller manager. Is that the one you are referring to?

Comment (liggitt) [reply cmd/kube-controller-manager/app/options/options_test.go]: oh, I meant a comment explaining what the `[]string{""}, []string{""}` args were, but I see those were copied from the existing test

**Pull request merged**